internal/pkg/agent/vault/vault.go

```diff
@@ -0,0 +1,38 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package vault
+
+import (
+	"context"
+	"runtime"
+	"time"
+)
+
+const (
+	// defaultFlockRetryDelay default file lock retry delay
+	defaultFlockRetryDelay = 10 * time.Millisecond
+
+	// lock file name
+	lockFile = `.lock`
+)
+
+type Vault interface {
+	Exists(ctx context.Context, key string) (bool, error)
+	Get(ctx context.Context, key string) (dec []byte, err error)
+	Set(ctx context.Context, key string, data []byte) (err error)
+	Remove(ctx context.Context, key string) (err error)
+	Close() error
+}
+
+func New(ctx context.Context, opts ...OptionFunc) (Vault, error) {
+	options := ApplyOptions(opts...)
+
+	if runtime.GOOS == "darwin" && !options.unprivileged {
+		// TODO add checks for unprivileged and proper fallback
```

Comment:
Is it done?

Reply:
Yes, yes it is :sweat_smile:

Reply:
Addressed in d3af9f9d69b4be829de6bb302445d2a95127106b